lib/broadway_kafka/producer.ex

```diff
@@ -215,6 +216,19 @@ defmodule BroadwayKafka.Producer do
     * `[:broadway_kafka, :assignments_revoked, :start | :stop | :exception]` spans -
       these events are emitted in "span style" when receiving assignments revoked call from consumer group coordinator
       See `:telemetry.span/3`.
+
+  ## Shared Client Performance
+
+  Enable shared client may drastically decrease performance. This happens because mutiple producers may block
+  each other waiting for the client response since the connection is hidden inside a process it becomes a
+  bottleneck.
```

Comment:
```suggestion
  Enabling shared client may drastically decrease performance. Since connection is handled by a single process,
  producers may block each other waiting for the client response.
```